config/config.go

```
@@ -1,25 +1,110 @@
 }
 
-// NewDefaultConfig returns a default configuration for connecting to the SPV Wallet API,
-// setting a one-minute timeout, using the default HTTP transport, and applying the
-// base API address as the addr value.
-func NewDefaultConfig(addr string) Config {
-	return Config{
-		Addr:      addr,
-		Timeout:   1 * time.Minute,
-		Transport: http.DefaultTransport,
+// setDefaultValues assigns default values to fields that are not explicitly set.
+func (cfg *Config) setDefaultValues() {
+	if cfg.Addr == "" {
+		cfg.Addr = "http://localhost:3003"
+	}
+	if cfg.Timeout == 0 {
+		cfg.Timeout = 1 * time.Minute
+	}
+	if cfg.Transport == nil {
+		cfg.Transport = http.DefaultTransport
+	}
+}
+
+// Option defines a function signature for modifying a Config.
+type Option func(*Config)
+
+// WithAddr sets the address in the configuration.
+func WithAddr(addr string) Option {
+	return func(cfg *Config) {
+		cfg.Addr = addr
+	}
+}
+
+// WithTimeout sets the timeout duration in the configuration.
+func WithTimeout(timeout time.Duration) Option {
+	return func(cfg *Config) {
+		cfg.Timeout = timeout
+	}
+}
+
+// WithTransport sets the HTTP transport in the configuration.
+func WithTransport(transport http.RoundTripper) Option {
+	return func(cfg *Config) {
+		cfg.Transport = transport
+	}
+}
+
+// NewConfig creates a new Config instance with optional customizations.
+func NewConfig(options ...Option) Config {
+	cfg := Config{}
+	for _, opt := range options {
+		opt(&cfg)
+	}
+	cfg.setDefaultValues()
+	return cfg
+}
+
+// loadConfigFromYAML loads configuration values from a YAML file using Viper.
+func loadConfigFromYAML(filePath string) (Config, error) {
+	viper.SetConfigFile(filePath)
+
+	if err := viper.ReadInConfig(); err != nil {
+		log.Printf("Error reading config file: %v", err)
+		return NewConfig(), err
+	}
+
+	viper.SetDefault("addr", "http://localhost:3003")
```

I think that setting the defaults after reading the config is a bit misleading... 
Not sure if I understand this func completely.